File: itsm-api/API/itsm/apps/tickets/serializers/comment.py
from rest_framework import serializers
from ..models.comment import Comment
from ...users.serializers.user import UserSerializer


class CommentSerializer(serializers.ModelSerializer):
    author = UserSerializer(read_only = True)

    class Meta:
        model = Comment
        fields = [
            'id',
            'author',
            'comment_type',
            'text',
            'is_internal',
            'created_at'
        ]

        read_only_fields = [
            'id',
            'comment_type',
            'author',
            'created_at'
        ]

File: itsm-front/my-app/src/interfaces/entities/Comment.ts
import type IUser from "./User";

export default interface IComment {
    id: number,
    ticket: string,
    author: IUser,
    text: string,
    created_at: string,
    comment_type: "comment" | "close_comment" | "cancel_comment" | "resolve_comment" | "reopen_comment" | "assign_comment" | "unassign_comment",
    is_internal?: boolean,
}
